commands/active_test.go

```diff
@@ -0,0 +1,51 @@
+package commands
+
+import (
+	"os"
+	"testing"
+
+	"github.com/docker/machine/drivers/fakedriver"
+	"github.com/docker/machine/libmachine/host"
+	"github.com/docker/machine/libmachine/state"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	h = host.Host{
+		Name:       "foo",
+		DriverName: "fakedriver",
+		Driver: &fakedriver.Driver{
+			MockState: state.Running,
+		},
+	}
+)
+
+func TestIsActiveReturnsFalseGivenNoActiveHost(t *testing.T) {
```

Comment:
> How do you mean? I find it groups related tests in a really concise, readable fashion.

_Maybe_, but it adds complexity. As soon as you have a `for` loop or `if` block, the code is now twice (or many) times more complex. Unit tests are a different kind of code than actual code, especially since there's no unit tests to test the unit tests...

> I don't understand. Tests always fail when the expected outputs don't match the actual outputs. What's ambiguous about that?

In my `go test` output, if I get a failure that says something like `TestIsActive failed` I have no immediate idea _why_ it failed. If, however, the failure says `TestIsActiveReturnsFalseGivenNoActiveHost failed`, I have a better idea of _what exact behaviour_ is failing. If I have multiple tests in a single method, I then have to start putting more information in the assertion, which is information that should be encoded in the function's name.

> A failing assertion doesn't prevent further tests from being run AFAICT

Ah. That's news to me... Most unit testing frameworks in most other languages I've worked with will immediately abort the test on the first failure. On further inspection, it looks like `testing` _does_ behave like you say.

But, the fact remains, it's easier to tell _what exactly_ is wrong when there's a 1:1 relationship with the assertion and the test function. For example:

```go
package main

import (
    "github.com/stretchr/testify/assert"
    "testing"
)

func Foo(i int) int {
    return i
}

func TestFoo(t *testing.T) {
    assert.Equal(t, Foo(1), 1)
    assert.Equal(t, Foo(2), 1)
    assert.Equal(t, Foo(3), 3)
    assert.Equal(t, Foo(4), 1)
}
```

```console
$ go test .
--- FAIL: TestFoo (0.00s)
        Error Trace:    main_test.go:14
	Error:  	Not equal: 2 (expected)
			        != 1 (actual)
		
        Error Trace:    main_test.go:16
	Error:  	Not equal: 4 (expected)
			        != 1 (actual)
		
FAIL
FAIL	_/tmp	0.007s
```

This gives me precious little information about _why_ those two tests failed. I'd have to use something like `assert.Equal(t, Foo(4), 1, "Foo(4)")` at the very least to even know what that assertion was actually _for_.

Anyway, all this is pretty academic, and while I feel strongly about it, this is your code, not mine.

If you're unconvinced, let me know and I'll change the test :)